exporters/autoexport/metrics.go

```diff
@@ -28,10 +28,17 @@ type MetricOption = option[metric.Reader]
 
 // WithFallbackMetricReader sets the fallback exporter to use when no exporter
 // is configured through the OTEL_METRICS_EXPORTER environment variable.
-func WithFallbackMetricReader(metricReaderFactory func(ctx context.Context) (metric.Reader, error)) MetricOption {
+func WithFallbackMetricReader(metricReaderFactory func(ctx context.Context, cfg config[metric.Reader]) (metric.Reader, error)) MetricOption {
 	return withFallbackFactory[metric.Reader](metricReaderFactory)
 }
 
+// WithProducer registers producers as an external Producer of metric data for this Reader.
+func WithProducer(producer metric.Producer) MetricOption {
```

Comment:
@open-telemetry/go-approvers Are we OK mixing option-based config with env-var-based config?  The alternative would be to have a registry for metric producers, and an environment variable that can set them (e.g. `OTEL_METRIC_PRODUCERS=opencensus,prometheus,goruntime`).  I would lean towards a WithProducer option (this PR) because producers seem tightly coupled to your instrumentation (e.g. opencensus or prometheus libraries), rather than export configuration.  But it does also break from the current "all env vars" model.

Reply:
Moreover, `OTEL_METRICS_PRODUCERS` is NOT defined in https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md.

Reply:
SIG meeting:
Let's go for env-var-based config and concurrently propose `OTEL_METRICS_PRODUCERS` in the spec.